pkg/csi/service/common/vsphereutil.go

```diff
@@ -909,39 +943,39 @@ func getDatastoreMoRefs(datastores []*vsphere.DatastoreInfo) []vim25types.Manage
 	return datastoreMoRefs
 }
 
-// Helper function to get Datastore object for given datastoreURL in the given
+// Helper function to get DatastoreInfo object for given datastoreURL in the given
 // virtual center.
-func getDatastoreObj(ctx context.Context, vc *vsphere.VirtualCenter,
-	datastoreURL string) (*vsphere.Datastore, error) {
+func getDatastoreInfoObj(ctx context.Context, vc *vsphere.VirtualCenter,
+	datastoreURL string) (*vsphere.DatastoreInfo, error) {
 	log := logger.GetLogger(ctx)
 	datacenters, err := vc.GetDatacenters(ctx)
 	if err != nil {
 		return nil, err
 	}
-	var datastoreObj *vsphere.Datastore
+	var datastoreInfoObj *vsphere.DatastoreInfo
 	for _, datacenter := range datacenters {
-		datastoreObj, err = datacenter.GetDatastoreByURL(ctx, datastoreURL)
+		datastoreInfoObj, err = datacenter.GetDatastoreInfoByURL(ctx, datastoreURL)
 		if err != nil {
 			log.Warnf("failed to find datastore with URL %q in datacenter %q from VC %q. Error: %+v",
 				datastoreURL, datacenter.InventoryPath, vc.Config.Host, err)
 		} else {
-			return datastoreObj, nil
+			return datastoreInfoObj, nil
 		}
 	}
 
 	return nil, logger.LogNewErrorf(log,
 		"Unable to find datastore for datastore URL %s in VC %+v", datastoreURL, vc)
 }
 
-// Helper function to get DatastoreMoRef for given datastoreURL in the given
+// Helper function to get DatastoreInfo object for given datastoreURL in the given
 // virtual center.
-func getDatastore(ctx context.Context, vc *vsphere.VirtualCenter,
-	datastoreURL string) (vim25types.ManagedObjectReference, error) {
-	dsObj, err := getDatastoreObj(ctx, vc, datastoreURL)
+func getDatastoreInfo(ctx context.Context, vc *vsphere.VirtualCenter,
```

Comment:
nit: This again is subjective but we can get rid of this method and use `getDatastoreInfoObj` wherever applicable, as it's not adding any additional functionality on top of that. We can have appropriate nil checks in calling places.

Reply:
Done.